src/util/equals.ts

```diff
@@ -0,0 +1,38 @@
+import { Value } from "../data/value";
+
+/**
+ * Compares two values for deep equality, including arrays, objects,
+ * and SurrealQL value types.
+ *
+ * @param x The first value to compare
+ * @param y The second value to compare
+ * @returns Whether the two values are recursively equal
+ */
+export function equals(x: unknown, y: unknown): boolean {
+	if (Object.is(x, y)) return true;
+	if (x instanceof Date && y instanceof Date) {
+		return x.getTime() === y.getTime();
+	}
+	if (x instanceof RegExp && y instanceof RegExp) {
+		return x.toString() === y.toString();
+	}
+	if (x instanceof Value && y instanceof Value) {
+		return x.equals(y);
+	}
+	if (
+		typeof x !== "object" ||
+		x === null ||
+		typeof y !== "object" ||
+		y === null
+	) {
+		return false;
```

Comment:
This includes all basic values right? strings, null, undefined, numbers, etc
```suggestion
		return x === y;
```

Reply:
Primitives, null, and undefined are already handled by the `Object.is` preceding it